<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>An Overview of the aqp Package</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>







<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">An Overview of the aqp Package</h1>



<p>Soil morphology, observed properties, and geomorphic context
represent a complex package of interrelated information that can be
difficult to analyze and communicate as a whole. Graphical methods such
as soil profile sketches and cross-sections represent a few of the
possible methods commonly used to report on these kind of data. The
Algorithms for Quantitative Pedology (AQP) project encompasses several
related R packages tailored to this style of work. A specialized data
structure (SoilProfileCollection) maintains linkages between soil
horizons, diagnostic features, above-ground data, and geomorphic
context. SoilProfileCollection objects can be filtered, subset,
resampled (over new depth intervals), and re-ordered; all while
preserving links to above and below-ground, linked data. Functions are
provided for the conversion of soil colors to and from Munsell notation
and several other color space coordinates. Graphical methods for the
SoilProfileCollection provide a simple but flexible framework for the
design and layout of soil profile sketches, aligned to x and or y axes
defined by linked data.</p>
<div id="related-tutorials-by-topic" class="section level1">
<h1>Related Tutorials by Topic</h1>
<div id="soil-morphology" class="section level2">
<h2>Soil Morphology</h2>
<ul>
<li><a href="https://ncss-tech.github.io/AQP/aqp/sketches.html">Soil
Profile Sketches</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/hz-boundaries.html">Visualization
of Horizon Boundaries</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/gen-hz-assignment.html">Assigning
Generalized Horizon Labels</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/dealing-with-bad-data.html">Dealing
With Troublesome Data</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/perturb.html">Simulation of
Soil Morphology by Perturbation</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/soiltexture-vizualization-ideas.html">Soil
Texture Visualization Ideas</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/SPC-plotting-ideas.html">SPC
Sketch Tips</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/vertical_vs_perpendicular.html">Vertical
vs.Â Perpendicular Horizon Depths</a></li>
</ul>
</div>
<div id="soil-color" class="section level2">
<h2>Soil Color</h2>
<ul>
<li><a href="https://ncss-tech.github.io/AQP/aqp/mix-colors.html">Mixing
Munsell Colors</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/color-contrast.html">Soil
Color Contrast</a></li>
<li><a href="https://ncss-tech.github.io/AQP/sharpshootR/aggregate-soil-color.html">Some
ideas on summarizing soil color</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/color-tiles.html">Soil
Color Tiles</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/investigating-soil-color.html">Investigating
Soil Color</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/soil-color-signatures.html">Some
Ideas on Soil Color Signatures</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/series-color-TP-graph.html">Soil
Color Transition Probability</a></li>
</ul>
</div>
<div id="numerical-classification" class="section level2">
<h2>Numerical Classification</h2>
<ul>
<li><a href="https://ncss-tech.github.io/AQP/aqp/genhz-distance-eval.html">Pair-Wise
Distances by Generalized Horizon Labels</a></li>
<li><a href="https://ncss-tech.github.io/AQP/aqp/profile-compare-factors.html">Pair-Wise
Distance Between Profiles and Mixed Variable Types</a></li>
</ul>
</div>
<div id="soil-hydrology" class="section level2">
<h2>Soil Hydrology</h2>
<ul>
<li><a href="https://ncss-tech.github.io/AQP/aqp/water-retention-curves.html">Theoretical
Water Retention Curves</a></li>
<li><a href="https://ncss-tech.github.io/AQP/soilDB/fetchKSSL-VG-demo.html">Water
Retention Curve Development from KSSL Data</a></li>
</ul>
</div>
<div id="geomorphology" class="section level2">
<h2>Geomorphology</h2>
<ul>
<li><a href="https://ncss-tech.github.io/AQP/sharpshootR/geomorphic-summaries-and-ordering.html">Hydrologic
Ordering of Geomorphic Proportions</a></li>
<li><a href="https://ncss-tech.github.io/AQP/soilDB/exploring-geomorph-summary.html">Exploring
Geomorphic Summaries</a></li>
</ul>
</div>
<div id="soil-taxonomy" class="section level2">
<h2>Soil Taxonomy</h2>
<ul>
<li><a href="https://ncss-tech.github.io/AQP/soilDB/subgroup-series.html">What
does a subgroup look like?</a></li>
<li><a href="https://ncss-tech.github.io/AQP/soilDB/soil-series-query-functions.html">Querying
Soil Series Data</a></li>
</ul>
</div>
<div id="multivariate-summaries-of-soil-property-data" class="section level2">
<h2>Multivariate Summaries of Soil Property Data</h2>
<ul>
<li><a href="https://ncss-tech.github.io/AQP/aqp/L1-profiles.html">L1
Profiles</a></li>
<li><a href="https://ncss-tech.github.io/AQP/soilDB/KSSL-demo.html">Getting and
Comparing KSSL Data</a></li>
</ul>
</div>
<div id="soil-spectroscopy" class="section level2">
<h2>Soil Spectroscopy</h2>
<ul>
<li><a href="https://ncss-tech.github.io/AQP/aqp/VNIR-spectra-to-color.html">Soil
Colors from VNIR Spectra</a></li>
</ul>
</div>
</div>
<div id="function-index-by-topic" class="section level1">
<h1>Function Index by Topic</h1>
<div id="soilprofilecollection-objects" class="section level2">
<h2>SoilProfileCollection Objects</h2>
<div id="creation" class="section level3">
<h3>Creation</h3>
<ul>
<li><code>depths()</code>: init an SPC from <code>data.frame</code></li>
<li><code>site()</code>: set or add site-level attributes of an
SPC<br />
</li>
<li><code>quickSPC()</code>: quickly build an SPC from simple text
templates</li>
<li><code>random_profile()</code>: generate random SPC from suite of
depth functions</li>
</ul>
</div>
<div id="metadata" class="section level3">
<h3>Metadata</h3>
<ul>
<li><code>hzdesgnname()</code>: get/set column containing horizon
designations</li>
<li><code>hzDesgn()</code>: get vector of horizon designations</li>
<li><code>hztexclname()</code>: get/set column containing horizon
texture class</li>
<li><code>metadata()</code>: get/set SPC metadata (list)</li>
<li><code>hzID()</code>: get vector of horizon IDs</li>
<li><code>hzidname()</code>: get/set column containing horizon IDs</li>
<li><code>horizonDepths()</code> get/set columns containing horizon top
and bottom depths</li>
</ul>
</div>
<div id="properties" class="section level3">
<h3>Properties</h3>
<ul>
<li><code>length()</code>: number of profiles in a SPC</li>
<li><code>nrow()</code>: number of horizons in a SPC</li>
<li><code>names()</code>: list of horizon and site names</li>
<li><code>siteNames()</code>: site-level column names</li>
<li><code>horizonNames()</code>: horizon-level column names</li>
</ul>
</div>
<div id="subset" class="section level3">
<h3>Subset</h3>
<ul>
<li><code>glom()</code>: extract horizons based on overlap criteria
defined by point or interval</li>
<li><code>trunc()</code>: truncate SPC to given depth interval</li>
<li><code>subset()</code>: subset profiles based on logical
expressions</li>
<li><code>subsetHz()</code>: subset horizons based on logical
expressions</li>
<li><code>[</code>: <code>data.frame</code>-like subsetting of profiles
(i-index) and/or horizons (j-index)</li>
<li><code>[[</code>: access site or horizon-level columns by name</li>
<li><code>k-index expressions</code>: <code>.FIRST</code>,
<code>.LAST</code>, <code>.HZID</code>, <code>.NHZ</code></li>
</ul>
</div>
<div id="depth" class="section level3">
<h3>Depth</h3>
<ul>
<li><code>min()</code>: minimum bottom depth within a SPC</li>
<li><code>max()</code>: maximum bottom depth within a SPC</li>
<li><code>depthOf()</code>: generalized âdepth toâ based on REGEX
matching</li>
<li><code>minDepthOf()</code>: special case of
<code>depthOf()</code></li>
<li><code>maxDepthOf()</code>: special case of
<code>depthOf()</code></li>
<li><code>getSoilDepthClass()</code>: estimate soil depth based on REGEX
matching applied to horizon designation and associated depth class</li>
<li><code>aggregateSoilDepth()</code>: statistical estimation of soil
depth (REGEX matching of horizon designation) within groups of
profiles</li>
</ul>
</div>
<div id="utility" class="section level3">
<h3>Utility</h3>
<ul>
<li><code>combine()</code>, <code>c()</code>: combine multiple SPCs into
a single SPC</li>
<li><code>duplicate()</code>: duplicate profiles within a SPC</li>
<li><code>perturb()</code>: randomly adjust horizon thickness or depths
to simulate from a template SPC</li>
<li><code>warpHorizons()</code>: expand / contract horizon
thickness</li>
<li><code>harmonize()</code>: create new profiles within a SPC based
sets of related horizon-level data</li>
<li><code>hzAbove()</code>, <code>hzBelow()</code>: locate horizons
above or below some criteria</li>
<li><code>unique()</code>: determine uniqueness among profiles of an SPC
via MD5 hash</li>
<li><code>split()</code>: split SPC into list of SPCs based on grouping
factor</li>
<li><code>site()</code>: get site data as <code>data.frame</code></li>
<li><code>horizons()</code>: get horizon data as
<code>data.frame</code></li>
<li><code>replaceHorizons()</code>: replace horizon data</li>
<li><code>diagnostic_hz()</code>: get/set diagnostic features</li>
<li><code>restrictions()</code>: get/set restrictions</li>
<li><code>denormalize()</code>: convert site-level data into
horizon-level data via replication</li>
<li><code>compositeSPC()</code>: downgrade an SPC to list of site and
horizon-level data</li>
</ul>
</div>
<div id="iteration" class="section level3">
<h3>Iteration</h3>
<ul>
<li><code>profileApply()</code>: apply a function to each profile within
an SPC (slow but simple interface)</li>
<li><code>summarizeSPC()</code>: perform group-wise summaries over
profiles within an SPC</li>
<li><code>transform()</code>: modify a SPC using expressions that
operation on site or horizon-level data</li>
</ul>
</div>
<div id="change-of-support" class="section level3">
<h3>Change of Support</h3>
<ul>
<li><code>dice()</code>: convert SPC to 1 depth-unit intervals by
replication</li>
<li><code>slab()</code>: apply an aggregate function over groups within
a âdice()-edâ SPC</li>
<li><code>spc2mpspline()</code>: interface to equal-area spline fitting
from mpspline2 package</li>
<li><code>segment()</code>: generate segment labels for depth-weighted
aggregation</li>
<li><code>L1_profiles()</code>: create representative profiles via
multivariate median (L1 estimator)</li>
<li><code>slicedHSD()</code>: apply Tukeyâs HSD over groups within a
âdice()-edâ SPC</li>
</ul>
</div>
<div id="horizon-depth-logic" class="section level3">
<h3>Horizon Depth Logic</h3>
<ul>
<li><code>accumulateDepths()</code>: fix horizon depths when old-style O
horizon notation has been used</li>
<li><code>fillHzGaps()</code>: fill topological gaps in horizon
depth</li>
<li><code>repairMissingHzDepths()</code>: attempt fixing missing or
duplicated horizon bottom depths</li>
<li><code>flagOverlappingHz()</code>: flag horizons with perfect
overlap</li>
<li><code>checkHzDepthLogic()</code>: apply battery of horizon depth
topological tests</li>
<li><code>splitLogicErrors()</code>: split an SPC according to variety
of possibly horizon depth errors</li>
<li><code>HzDepthLogicSubset()</code>: remove profiles from an SPC if
any depth logic errors are present</li>
</ul>
</div>
<div id="data-qc" class="section level3">
<h3>Data QC</h3>
<ul>
<li><code>evalMissingData()</code>: report metrics of missing data by
profile within SPC</li>
<li><code>missingDataGrid()</code>: visual indication of missing
data</li>
<li><code>profileInformationIndex()</code>: experimental indices of
âinformation contentâ by profile</li>
</ul>
</div>
<div id="object-coercion" class="section level3">
<h3>Object Coercion</h3>
<ul>
<li><code>as(SPC, &#39;list&#39;)</code>: convert SPC to <code>list</code></li>
<li><code>as(SPC, &#39;data.frame&#39;)</code>: convert site and horizon data to
<code>data.frame</code></li>
<li><code>as(SPC, &#39;sf&#39;)</code>: convert site and spatial data to sf
object</li>
</ul>
</div>
<div id="spatial-data" class="section level3">
<h3>Spatial Data</h3>
<ul>
<li><code>prj()</code>: get/set coordinate reference system (CRS)
metadata</li>
<li><code>initSpatial()</code>: set (site-level) column names containing
coordinates</li>
<li><code>getSpatial()</code>: get spatial data (site + coordinates)
from an SPC</li>
</ul>
</div>
<div id="internal-consistency" class="section level3">
<h3>Internal Consistency</h3>
<ul>
<li><code>checkSPC()</code>: check SPC for internal consistency</li>
<li><code>rebuildSPC()</code>: re-make an SPC from a previous version of
aqp (rarely required)</li>
</ul>
</div>
</div>
<div id="soil-profile-sketches" class="section level2">
<h2>Soil Profile Sketches</h2>
<ul>
<li><code>plotSPC()</code>: create sketches of soil profiles from an
SPC</li>
<li><code>explainPlotSPC()</code>: explain various elements of a soil
profile sketch</li>
<li><code>groupedProfilePlot()</code>: create grouped sketches of soil
profiles</li>
<li><code>plotMultipleSPC()</code>: create sketches from multiple,
possibly non-conformal SPCs</li>
<li><code>addBracket()</code>: add vertical brackets beside soil profile
sketches</li>
<li><code>addDiagnosticBracket()</code>: add vertical brackets beside
soil profile sketches based on diagnostic features</li>
<li><code>addVolumeFraction()</code>: add visual explanation of coarse
fragment volume to soil profile sketches</li>
<li><code>alignTransect()</code>: align soil profiles to an external
gradient such as topographic sequence</li>
</ul>
</div>
<div id="soil-color-color-science" class="section level2">
<h2>Soil Color / Color Science</h2>
<div id="color-conversion" class="section level3">
<h3>Color Conversion</h3>
<ul>
<li><code>col2Munsell()</code>: convert various color notations to
Munsell notation</li>
<li><code>munsell2rgb()</code>: convert Munsell notation to sRGB or
CIELAB color coordinates</li>
<li><code>parseMunsell()</code>: parse and optionally convert a munsell
color</li>
<li><code>spec2Munsell()</code>: estimate the closest Munsell color
given reflectance spectra in the visible range</li>
<li><code>getClosestMunsellChip()</code>: estimate a reasonably close
Munsell color given non-standard notation</li>
<li><code>estimateSoilColor()</code>: estimate moist soil color from dry
soil color (and vice versa)</li>
</ul>
</div>
<div id="comparison" class="section level3">
<h3>Comparison</h3>
<ul>
<li><code>colorContrast()</code>: pair-wise color contrast and CIE2000
(dE00) based on colors in Munsell notation</li>
<li><code>colorContrastPlot()</code>: visual explanation of soil color
contrast and dE00</li>
<li><code>contrastChart()</code>: Munsell color book style explanation
of soil color contrast and dE00</li>
<li><code>soilColorSignature()</code>: derive soil color signatures for
profiles within an SPC</li>
</ul>
</div>
<div id="aggregation" class="section level3">
<h3>Aggregation</h3>
<ul>
<li><code>colorChart()</code>: Munsell color book representation of
color frequency</li>
<li><code>aggregateColor()</code>: estimate color proportions within an
SPC according within groups of horizons</li>
<li><code>colorQuantiles()</code>: marginal and L1 quantiles of color in
CIELAB coordinates</li>
</ul>
</div>
<div id="utility-1" class="section level3">
<h3>Utility</h3>
<ul>
<li><code>huePosition()</code>: generate an ordered factor of the
standard Munsell hues</li>
<li><code>huePositionCircle()</code>: graphical representation of the
standard Munsell hues, with optional simulation of common color vision
deficiency</li>
<li><code>simulateColor()</code>: simulate a range of Munsell colors
given measures of central tendency and spread</li>
<li><code>previewColors()</code>: graphical preview of colors as a grid
or via nMDS</li>
<li><code>soilPalette()</code>: generate swatch-like arrangements of
colors and labels</li>
<li><code>equivalentMunsellChips()</code>: for a specified Munsell
color, identify other Munsell colors with a very lower CIE2000 color
contrast difference</li>
</ul>
</div>
<div id="simulation-of-mixtures" class="section level3">
<h3>Simulation of Mixtures</h3>
<ul>
<li><code>mixMunsell()</code>: simulate an subtractive mixture of
pigments specified in Munsell notation</li>
<li><code>plotColorMixture()</code>: simulate a subtractive mixture of
pigments, display reflectance spectra</li>
</ul>
</div>
</div>
<div id="numerical-classification-of-soil-profiles" class="section level2">
<h2>Numerical Classification of Soil Profiles</h2>
<ul>
<li><code>NCSP()</code>: numerical classification of soil profiles,
within a <code>SoilProfileCollection</code></li>
<li><code>compareSites()</code>: pair-wise comparison of site-level data
from a <code>SoilProfileCollection</code></li>
</ul>
</div>
<div id="pedology" class="section level2">
<h2>Pedology</h2>
<ul>
<li><code>allocate()</code>: perform one of several classification
systems to soil property data</li>
<li><code>estimateAWC()</code>: estimate plant-available water holding
capacity</li>
<li><code>correctAWC()</code>: apply rock fragment or soluble salt
corrections to AWC estimates</li>
<li><code>hzDistinctnessCodeToOffset()</code>: convert horizon boundary
distinctness codes to vertical offset</li>
<li><code>hzTopographyCodeToLineType()</code>: convert horizon boundary
topography codes to line types</li>
<li><code>hzTopographyCodeToOffset()</code>: convert horizon boundary
topography codes to offset</li>
<li><code>ph_to_rxnclass()</code>: convert pH to reaction class</li>
<li><code>rxnclass_to_ph()</code>: convert reaction class to pH
range</li>
<li><code>ReactionClassLevels()</code>: ordered factor of reaction
classes</li>
<li><code>horizonColorIndices()</code>: compute various soil color-based
indices to horizons within an SPC</li>
</ul>
<div id="soil-texture" class="section level3">
<h3>Soil Texture</h3>
<ul>
<li><code>textureTriangleSummary()</code>: graphical summary of sand,
silt, clay fractions on a soil texture triangle<br />
</li>
<li><code>bootstrapSoilTexture()</code>: simulation of realistic
compositions (sand, silt, clay) from a small set of example data</li>
<li><code>SoilTextureLevels()</code>: ordered factor of soil texture
classes</li>
<li><code>texcl_to_ssc()</code>: convert soil texture classes to sand,
silt, clay centroids</li>
<li><code>ssc_to_texcl()</code>: convert sand, silt, clay values to soil
texture class</li>
<li><code>texture_to_taxpartsize()</code>: convert soil texture to Soil
Taxonomy particle size class</li>
</ul>
</div>
<div id="coarse-fragments" class="section level3">
<h3>Coarse Fragments</h3>
<ul>
<li><code>fragmentSieve()</code>: classify coarse fragments by fragment
diameter</li>
<li><code>texmod_to_fragvoltot()</code>: estimate ranges in coarse
fragment volume based on a soil texture modifier</li>
<li><code>texture_to_texmod()</code></li>
<li><code>fragvol_to_texmod()</code></li>
<li><code>fragmentClasses()</code>: coarse fragment diameter thresholds
used by USDA-NRCS</li>
</ul>
</div>
<div id="soil-taxonomy-1" class="section level3">
<h3>Soil Taxonomy</h3>
<ul>
<li><code>getArgillicBounds()</code>: estimate the upper and lower
boundaries of an argillic horizon</li>
<li><code>getCambicBounds()</code>: estimate the upper and lower
boundaries of a cambic horizon</li>
<li><code>getSurfaceHorizonDepth()</code></li>
<li><code>getMineralSoilSurfaceDepth()</code></li>
<li><code>getPlowLayerDepth()</code></li>
<li><code>hasDarkColors()</code></li>
<li><code>estimatePSCS()</code></li>
</ul>
</div>
<div id="generalized-horizon-labels-ghl" class="section level3">
<h3>Generalized Horizon Labels (GHL)</h3>
<ul>
<li><code>generalize.hz()</code>: apply REGEX rules to group horizon
designations into a reduced set of âgeneralized horizon labelsâ</li>
<li><code>evalGenHZ()</code>: evaluate internal consistency of assigned
GHL</li>
<li><code>genhzTableToAdjMat()</code>: convert a cross-tabulation of GHL
vs.Â original horizon designations to adjacency matrix</li>
<li><code>get.ml.hz()</code>: extract most likely horizon boundary
depths from probability depth functions</li>
<li><code>guessGenHzLevels()</code>: estimate the correct ordering of
GHL given horizon depths</li>
<li><code>GHL()</code>: get/set GHL metadata for a
<code>SoilProfileCollection</code></li>
</ul>
</div>
</div>
<div id="misc." class="section level2">
<h2>Misc.</h2>
<ul>
<li><code>invertLabelColor()</code>: automatic adjustment of label color
for maximum contrast, based on specified background color</li>
<li><code>hzTransitionProbabilities()</code>: derive transition
probability matrix from horizon level data</li>
<li><code>mostLikelyHzSequence()</code>: use Markov Chains to predict
the most likely sequence of horizons</li>
</ul>
<div id="accuracy-and-uncertainty" class="section level3">
<h3>Accuracy and Uncertainty</h3>
<ul>
<li><code>shannonEntropy()</code>: Shannon entropy</li>
<li><code>brierScore()</code>: Brierâs score</li>
<li><code>tauW()</code>: weighted tau statistic</li>
</ul>
</div>
<div id="overlapping-annotation" class="section level3">
<h3>Overlapping Annotation</h3>
<ul>
<li><code>findOverlap()</code>: identify overlap within a vector of
positions based on a given threshold</li>
<li><code>overlapMetrics()</code>: metrics of overlap within a vector or
positions based on a given threshold</li>
<li><code>fixOverlap()</code>: attempt the minimum of adjustments to
vector of positions such that a given distance threshold is
enforced</li>
</ul>
</div>
</div>
<div id="example-data" class="section level2">
<h2>Example Data</h2>
<ul>
<li><code>sp1</code></li>
<li><code>sp2</code></li>
<li><code>sp3</code></li>
<li><code>sp4</code></li>
<li><code>sp5</code></li>
<li><code>sp6</code></li>
<li><code>sierraTransect</code></li>
<li><code>wilson2022</code></li>
<li><code>rowley2019</code></li>
<li><code>jacobs2000</code></li>
<li><code>osd</code></li>
<li><code>SPC.with.overlap</code></li>
<li><code>us.state.soils</code></li>
<li><code>soil_minerals</code></li>
<li><code>munsell</code></li>
<li><code>equivalent_munsell</code></li>
<li><code>munsellHuePosition</code></li>
<li><code>munsell.spectra</code></li>
<li><code>spectral.reference</code></li>
<li><code>ROSETTA.centroids</code></li>
<li><code>reactionclass</code></li>
<li><code>soiltexture</code></li>
</ul>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
